gravitee-gateway-handlers/gravitee-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/spring/ApiPropertiesEncryptorConfiguration.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.handlers.api.spring;
+
+import io.gravitee.common.util.DataEncryptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.env.Environment;
+
+/**
+ * @author GraviteeSource Team
+ */
+public class ApiPropertiesEncryptorConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApiPropertiesEncryptorConfiguration.class);
+    private static final String DEFAULT_API_PROPERTIES_ENCRYPTION_SECRET = "vvLJ4Q8Khvv9tm2tIPdkGEdmgKUruAL6";
+
+    private String apiPropertiesEncryptionSecret;
+
+    @Autowired
+    private Environment environment;
+
+    @Bean
+    public DataEncryptor apiPropertiesEncryptor() {
+        apiPropertiesEncryptionSecret =
+            environment.getProperty("apiProperties.encryption.secret", DEFAULT_API_PROPERTIES_ENCRYPTION_SECRET);
+        if (DEFAULT_API_PROPERTIES_ENCRYPTION_SECRET.equals(apiPropertiesEncryptionSecret)) {
+            LOGGER.warn("");
+            LOGGER.warn("##############################################################");
+            LOGGER.warn("#                      SECURITY WARNING                      #");
+            LOGGER.warn("##############################################################");
+            LOGGER.warn("");
+            LOGGER.warn("You still use the default api properties encryption secret.");
+            LOGGER.warn("This known secret can be used to decrypt your properties.");
+            LOGGER.warn("Please customize the 'apiProperties.encryption.secret' parameter value, or ask your administrator to do it.");
+            LOGGER.warn("");
+            LOGGER.warn("##############################################################");
+            LOGGER.warn("");
```

Comment:
Do you think it could be moved inside the `DataEncryptor` ? That way, it will not be forgotten and if we decide to change the logic it will require less efforts (ex: generate and display the secret for the first start, stop the start if not changed, ...).